storage/series/series.go

```diff
@@ -180,8 +185,18 @@ func (s *dbSeries) updateBlocksWithLock() updateBlocksResult {
 			case CacheRecentlyRead:
 				sinceLastRead := now.Sub(currBlock.LastReadTime())
 				shouldUnwire = sinceLastRead >= wiredTimeout
+			case CacheLRU:
+				// The tick is responsible for managing the lifecycle of blocks that were not
+				// read from disk (not retrieved), and the WiredList will manage those that were
+				// retrieved from disk.
+				if !currBlock.WasRetrievedFromDisk() {
+					shouldUnwire = true
```

Comment:
Technically speaking, if you're in the `IsBlockRetrievable(...)` code path then it definitely is retrievable from disk.  So instead of doing this check we can probably just do `shouldUnwire = true` here.

Reply:
is that 100% true? Seems like something could be retrievable because it was flushed to disk, but still did not originate from disk because the tick that evicts it post-flush hasn't occurred yet

Reply:
@r let me know if you agree with me, and if so I'll add a comment

Reply:
Ah yes, ok I see so you don't want to unwire blocks brought from disk because the LRU does that.

Reply:
Hey so what about the blocks that get into the wired list just after being rotated into the series from the buffer when they are read (with the ReadEncoded and then the series `OnReadBlock(...)` call before a flush occurs.

Reply:
Good call on the nit, I'll adjust that.

Regarding your other comment, blocks that are created by rotating into the series from the buffer will never get into the WiredList.

See the OnReadBlock function on the Series

```
// OnReadBlock is only called for blocks that were read from memory, regardless of
// whether the data originated from disk or buffer rotation.
func (s *dbSeries) OnReadBlock(b block.DatabaseBlock) {
	if list := s.opts.DatabaseBlockOptions().WiredList(); list != nil {
		// The WiredList is only responsible for managing the lifecycle of blocks
		// retrieved from disk.
		if b.WasRetrievedFromDisk() {
			// Need to update the WiredList so it knows which blocks have been
			// most recently read.
			list.Update(b)
		}
	}
}
```

This simplifies things a lot at the cost that after rotation we basically force these blocks to be read from disk, even if they were being read periodically while they were in the buffer. In practice this worked fine from a performance perspective.

Reply:
Anything else and it gets too messy I think